servers/src/main/java/tachyon/worker/block/evictor/LRUEvictor.java

```diff
@@ -0,0 +1,128 @@
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.worker.block.evictor;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.google.common.base.Optional;
+
+import tachyon.Pair;
+import tachyon.worker.BlockStoreLocation;
+import tachyon.worker.block.BlockAccessEventListener;
+import tachyon.worker.block.BlockMetadataManager;
+import tachyon.worker.block.meta.BlockMeta;
+
+/**
+ * Not thread-safe
+ */
+public class LRUEvictor extends Evictor implements BlockAccessEventListener {
+  /** Double-Link List, most recently accessed block is at tail of the list */
+  private class Node {
+    long blockId;
+    Node prev, next;
+    Node() {
+      blockId = -1;
+      prev = next = null;
+    }
+    Node(long blkId) {
+      blockId = blkId;
+      prev = next = null;
+    }
+
+    void append(Node node) {
+      node.next = next;
+      node.prev = this;
+      next = node;
+    }
+
+    // Remove itself from the List
+    void remove() {
+      if (prev != null) {
+        prev.next = next;
+      }
+      if (next != null) {
+        next.prev = prev;
+      }
+      next = prev = null;
+    }
+  }
+  private Node mHead;
+  private Node mTail;
+  /** Map from blockId to corresponding Node in the Double-Link List */
+  private Map<Long, Node> mCache;
+
+
+  public LRUEvictor(BlockMetadataManager meta) {
+    super(meta);
+
+    mHead = new Node();
+    mTail = new Node();
+    mHead.append(mTail);
+    mCache = new HashMap<Long, Node>();
+  }
+
+  /**
+   * Free space in the given block store location. The location can be a specific location, or
+   * {@link BlockStoreLocation#anyTier()} or {@link BlockStoreLocation#anyDirInTier(int)} .
+   *
+   * If the total free space is fewer than {@param bytes}, they will all be evicted
+   *
+   * @param bytes the size in bytes
+   * @param location the location in block store
+   * @return an eviction plan to achieve the freed space
+   */
+  @Override
+  public EvictionPlan freeSpace(long bytes, BlockStoreLocation location) {
+    List<Pair<Long, BlockStoreLocation>> toMove = new ArrayList<Pair<Long, BlockStoreLocation>>();
+    List<Long> toEvict = new ArrayList<Long>();
+
+    Node p = mHead;
+    long evictBytes = 0;
+    while (p.next != mTail && evictBytes < bytes) {
+      Optional<BlockMeta> meta = mMeta.getBlockMeta(p.blockId);
+      if (!meta.isPresent() || !meta.get().getBlockLocation().belongTo(location)) {
+        continue;
+      }
+
+      evictBytes += meta.get().getBlockSize();
+      toEvict.add(p.blockId);
+
+      mCache.remove(p.blockId);
+
+      Node current = p;
+      p = p.next;
+      current.remove();
+    }
+
+    return new EvictionPlan(toMove, toEvict);
+  }
+
+  @Override
+  public void onAccessBlock(long userId, long blockId) {
```

Comment:
this method is not thread-safe. If multiple threads are calling onAccessBlock at the same time, you may have race condition.